hazelcast/src/main/java/com/hazelcast/query/impl/CachedQueryEntry.java

```diff
@@ -110,34 +107,44 @@ public Comparable getAttribute(String attributeName) throws QueryException {
     }
 
     private Object getTargetObject(boolean key) {
-        Object targetObject;
         if (key) {
-            if (keyObject == null) {
-                if (keyData.isPortable()) {
-                    targetObject = keyData;
-                } else {
-                    targetObject = getKey();
-                }
+            return getTargetKey();
+        } else {
+            return getTargetValue();
+        }
+    }
+
+    private Object getTargetValue() {
+        Object targetObject;
```

Comment:
You could safe some lines of code and indentation with direct `return` statements. Also the definition order of `getTargetValue()` and `getTargetKey()` methods should be the same as the invocation order.